src/core/DynamicPathTracingSceneGenerator.js

```diff
@@ -28,28 +28,39 @@ export class DynamicPathTracingSceneGenerator {
 		this.geometry = new BufferGeometry();
 		this.materials = null;
 		this.textures = null;
-		this.staticGeometryGenerator = new StaticGeometryGenerator( this.scene );
+		this.staticGeometryGenerator = new StaticGeometryGenerator( this.objects );
 
 	}
 
 	dispose() {}
 
 	generate() {
 
-		const { scene, staticGeometryGenerator, geometry } = this;
+		const { objects, staticGeometryGenerator, geometry } = this;
 		if ( this.bvh === null ) {
 
 			const attributes = [ 'position', 'normal', 'tangent', 'uv' ];
-			scene.traverse( c => {
+
+			function checkObject( c ) {
 
 				if ( c.isMesh ) {
 
-					const normalMapRequired = ! ! c.material.normalMap;
-					setCommonAttributes( c.geometry, { attributes, normalMapRequired } );
+					c.traverse( ( cTrav ) => {
+
+						if ( cTrav.isMesh ) {
+
+							const normalMapRequired = ! ! c.material.normalMap;
+							setCommonAttributes( c.geometry, { attributes, normalMapRequired } );
+
+						}
+
+					} );
 
 				}
 
-			} );
+			}
+
+			objects.forEach( checkObject );
```

Comment:
I prefer to use a for loop rather than forEach generally since they're faster:

```js
for ( let i = 0, l = objects.length; i < l; i ++ ) {

  const c = objects[ i ];

  // ...

}
```